plane/src/types/backend_state.rs

```diff
@@ -393,7 +427,8 @@ impl BackendStatusStreamEntry {
 
         let termination_kind = match state {
             BackendState::Terminated { termination, .. } => termination,
-            BackendState::Terminating { termination, .. } => Some(termination),
+            BackendState::Terminating { .. } => Some(TerminationKind::Soft),
+            BackendState::HardTerminating { .. } => Some(TerminationKind::Hard),
```

Comment:
Do we need to pull out the `termination_reason` for `HardTerminating` a few lines above, too?